src/AutorestSwift/View Models/ReturnTypeViewModel.swift

```diff
@@ -26,13 +26,22 @@
 
 import Foundation
 
+enum ReturnTypeEnum: String {
+    case pageable
+    case decoding
```

Comment:
The name `decoding` does not seem to consistent with `pageable` , `statuscode`.  `pageable` , `statuscode` matches the name of Enum 'ReturnType', but decoding seems to say what you are doing to do with the return type.

Reply:
I agree, I don't love these names. I could add doc comments to clarify or we could just choose different ones. Thoughts?

```Swift
/// returns PagedCollection<Codable>
case pageable
/// returns some Codable type
case decoding
/// no return data. Simply checks the status code
case statusCode
```

Reply:
So we aren't really trying to match the code model's ReturnType enum here. Most of those things would fall into the "decoding" bucket where we can just decode them with `JSONDecoder.decode(Codable.type, data: data)`. It's more trying to reflect the flavor of the code that needs to go in that section to accomplish the goal.

Reply:
maybe 'ResponseWithBody' ?

Reply:
I would be okay with that. I pushed one option, but I might like this better:
```
case pagedBody
case body
case noBody
```
Thoughts?

Reply:
Sound good!

Reply:
Or
```swift
enum BodyType: String {
case paged
case normal
case none
}
```

Reply:
i like better
```
case pagedBody
case body
case noBody
```

Reply:
Updated. Let me know if it looks good to you.

Reply:
Looks good.